scene/debugger/scene_debugger.cpp

```diff
@@ -221,7 +221,11 @@ void SceneDebugger::_save_node(ObjectID id, const String &p_path) {
 	ERR_FAIL_COND(!node);
 
 	HashMap<const Node *, Node *> duplimap;
+#ifdef TOOLS_ENABLED
```

Comment:
You don't need `duplimap` in template build, so it can go inside the `TOOLS_ENABLED`.

Though it's not used *at all*, so you could check if the method supports just passing a nullptr instead of allocating and filling a hashmap for nothing.

Reply:
It can't be null, because it's passed as reference.

Reply:
Looks like it still needs to be defined in the scope first to be passed as reference.

Reply:
But the same code is used in `duplicate_from_editor()` :/ (it's an overload for method that takes 2 hashmaps)